libraries/AP_Terrain/AP_Terrain.cpp

```diff
@@ -143,22 +147,30 @@ bool AP_Terrain::height_amsl(const Location &loc, float &height)
         home_loc = loc;
     }
 
+    // apply correction which assumes home altitude is at terrain altitude
+    if (corrected) {
+        height += (ahrs.get_home().alt * 0.01f) - home_height;
+    }
```

Comment:
Two points about this:

- the _corrected_ argument is only used in one place so I'm not sure if it is valuable to have it in the function (the caller could do it)
- I'm not understanding what this does. If home altitude is at terrain altitude then _home_height_ wouldn't be the same as _ahrs.get_home().alt_?